connect/json/src/test/java/org/apache/kafka/connect/json/JsonConverterTest.java

```diff
@@ -563,6 +563,13 @@ public void structToJson() {
                 converted.get(JsonSchema.ENVELOPE_PAYLOAD_FIELD_NAME));
     }
 
+    @Test
+    public void structSchemaIdentical() {
```

Comment:
I think this test will pass even without your change as you creating struct object using the same schema.

Reply:
@bartektartanus Isn't that the purpose of this test? To create a scenario in which a struct and schema is passed to JsonConverter#convertToJson()?

Reply:
Yes, but you are passing the same schema in the same OBJECT. So comparing them using == will return true. In my case the schema and schema in struct are the same (equal) but in different objects, co equals method returns true, but == returns false. 
While this test might be useful, it does not cover the change you are making here.